indra/llcharacter/llik.cpp

```diff
@@ -1384,7 +1402,8 @@ void LLIK::Joint::setWorldPos(const LLVector3& pos)
 // this should only be called once before starting IK algorithm iterations
 void LLIK::Joint::setLocalPos(const LLVector3& pos)
 {
-    mLocalPos = pos;
+    mLocalPos = pos.scaledVec(mLocalScale);
+    mLocalPosLength = mLocalPos.length();
```

Comment:
Here, and in a few other spots, we apply `mLocalScale `to `mLocalPos `(when either of them changes) before starting IK calculations.  This way we don't have to constantly recompute `mLocalPos` during IK.